Bio/PDB/PDBList.py

```diff
@@ -701,88 +702,166 @@ def get_seqres_file(self, savefile="pdb_seqres.txt"):
-            if with_assemblies:
-                # get all assembly structures
-                pl.download_all_assemblies(file_format=file_format)
+    Quick access to the structure lists on the PDB or its mirrors.
+    Requires a mode (update/all/obsol/assemb/single/multi) to be specified.
+    By default (no format specified) structures are downloaded as PDBx/mmCif files.
+    """
 
-        elif sys.argv[1] == "obsol":
-            # get all obsolete entries
-            pl.download_obsolete_entries(pdb_path, file_format=file_format)
+    # Instantiate argument parser/subparser
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.RawDescriptionHelpFormatter, description=desc
+    )
+
+    # Add positional argument for pdb_path
+    parser.add_argument(
+        "pdb_path",
+        nargs="?",
+        default=os.getcwd(),
+        help="local path to use, defaults to the current working directory",
+    )
+
+    # Add mutually exclusive arguments for mode
+    mode = parser.add_mutually_exclusive_group(required=True)
+    mode.add_argument(
+        "-update",
+        action="store_true",
+        help="write weekly PDB updates to local pdb tree",
+    )
+    mode.add_argument(
+        "-all", action="store_true", help="write all PDB entries to local pdb tree"
+    )
+    mode.add_argument(
+        "-obsol",
+        action="store_true",
+        help="write all obsolete PDB entries to local pdb tree",
+    )
+    mode.add_argument(
+        "-assemb",
+        action="store_true",
+        help="write all assemblies for each PDB entry to local pdb tree",
+    )
+    mode.add_argument(
+        "-single", help="retrieve single structure - accepts a single PDB code"
+    )
+    mode.add_argument(
+        "-multi",
+        help="retrieve a set of structures - accepts a comma separated list of PDB codes",
+    )
+
+    # Add mututally exclusive arguments for format
+    format = parser.add_mutually_exclusive_group()
+    format.add_argument(
+        "-pdb", action="store_true", help="downloads structures in PDB format"
+    )
+    format.add_argument(
+        "-xml", action="store_true", help="downloads structures in PDBML (XML) format"
+    )
+    format.add_argument(
+        "-mmtf", action="store_true", help="downloads structures in mmtf format"
+    )
+
+    # Add final options
+    others = parser.add_argument_group()
+    others.add_argument(
+        "-d",
+        action="store_true",
+        help="a single directory will be used as pdb_path, not a tree",
+    )
+    others.add_argument(
+        "-o", action="store_true", help="overwrite existing structure files"
+    )
+    others.add_argument(
+        "-with-assemblies",
+        action="store_true",
+        help="downloads assemblies along with regular entries",
+    )
+
+    # Parse arguments
+    args = parser.parse_args()
+
+    # Set path
+    pdb_path = args.pdb_path
+
+    # Instantiate PDBList class
+    pl = PDBList(pdb=pdb_path)
```

Comment:
What about a (or multiple) new optional arg to set the server?

Reply:
I think this is probably linked to #4012 and maybe needs to wait until that is resolved? Could certainly add more optional arguments but I don't want to alter the class in this PR.

Reply:
not necessarily, but as you feel :)